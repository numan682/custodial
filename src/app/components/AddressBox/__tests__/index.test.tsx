import * as React from 'react'
import copy from 'copy-to-clipboard'
import { screen } from '@testing-library/dom'
import { render } from '@testing-library/react'
import userEvent from '@testing-library/user-event'

import { AddressBox } from '../index'

jest.mock('copy-to-clipboard')

const testAddress = 'oasis1qqurxkgavtcjjytneumeclx59ds3avjaqg7ftqph'

const renderComponent = () => render(<AddressBox address={testAddress} />)

describe('<AddressBox />', () => {
  it('should render address properly', () => {
    const { container } = renderComponent()
    expect(container).toMatchSnapshot()
  })

  it('should be able to copy address to clipboard', async () => {
    renderComponent()
    await userEvent.click(screen.getByTestId('copy-address'))
    expect(copy).toHaveBeenCalledWith(testAddress)
  })

  it('should be able to show a notification', async () => {
    renderComponent()
    jest.mocked(copy).mockReturnValue(true) // Copy must return true so that the notification is actually displayed
    await userEvent.click(screen.getByTestId('copy-address'))
    expect(await screen.getByText('account.addressCopied')).toBeInTheDocument()
  })
})
