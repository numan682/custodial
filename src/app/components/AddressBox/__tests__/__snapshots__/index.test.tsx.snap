// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddressBox /> should render address properly 1`] = `
.c2 {
  display: inline-block;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 18px;
  height: 18px;
  fill: #666666;
  stroke: #666666;
}

.c2 g {
  fill: inherit;
  stroke: inherit;
}

.c2 *:not([stroke])[fill='none'] {
  stroke-width: 0;
}

.c2 *[stroke*='#'],.c2 *[STROKE*='#'] {
  stroke: inherit;
  fill: none;
}

.c2 *[fill-rule],
.c2 *[FILL-RULE],
.c2 *[fill*='#'],.c2 *[FILL*='#'] {
  fill: inherit;
  stroke: none;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  padding-right: 12px;
  border-radius: 5px;
}

.c3 {
  font-size: 18px;
  line-height: 24px;
  font-weight: bold;
  word-break: break-word;
}

.c4 {
  font-size: 18px;
  line-height: 24px;
}

.c1 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  color: inherit;
  outline: none;
  border: none;
  padding: 0;
  text-align: inherit;
  line-height: 0;
  padding: 12px;
}

.c1:focus {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c1:focus > circle,
.c1:focus > ellipse,
.c1:focus > line,
.c1:focus > path,
.c1:focus > polygon,
.c1:focus > polyline,
.c1:focus > rect {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c1:focus::-moz-focus-inner {
  border: 0;
}

.c1:focus:not(:focus-visible) {
  outline: none;
  box-shadow: none;
}

.c1:focus:not(:focus-visible) > circle,.c1:focus:not(:focus-visible) > ellipse,
.c1:focus:not(:focus-visible) > line,.c1:focus:not(:focus-visible) > path,
.c1:focus:not(:focus-visible) > polygon,.c1:focus:not(:focus-visible) > polyline,
.c1:focus:not(:focus-visible) > rect {
  outline: none;
  box-shadow: none;
}

.c1:focus:not(:focus-visible)::-moz-focus-inner {
  border: 0;
}

@media only screen and (max-width:768px) {
  .c0 {
    padding-right: 6px;
  }
}

<div>
  <div
    class="c0"
  >
    <button
      class="c1"
      data-testid="copy-address"
      type="button"
    >
      <svg
        aria-label="Copy"
        class="c2"
        viewBox="0 0 24 24"
      >
        <path
          d="M9 15h8-8zm0-4h10H9zm0-4h4-4zm7-6v6h6M6 5H2v18h16v-4m4 0H6V1h11l5 5v13z"
          fill="none"
          stroke="#000"
          stroke-width="2"
        />
      </svg>
    </button>
    <span
      class="c3"
    >
      <span
        class="c4"
        style="font-family: Roboto mono; letter-spacing: 0;"
      >
        <span
          class="notranslate"
          translate="no"
        >
          oasis1 qqur xkga vtcj jytn eume clx5 9ds3 avja qg7f tqph
        </span>
      </span>
    </span>
  </div>
</div>
`;
